src/plus/startWork/startWork.ts

```diff
@@ -0,0 +1,462 @@
+				if (button === StartWorkQuickInputButton) {
+					this.startWork(state, item);
+					return true;
+				}
+				return false;
+			},
+		});
+
+		const selection: StepSelection<typeof step> = yield step;
+		if (!canPickStepContinue(step, state, selection)) {
+			return StepResultBreak;
+		}
+		const element = selection[0];
+		return { ...element.item };
+	}
+
+	private *confirmStep(state: StartWorkStepState, _context: Context): StepResultGenerator<StartWorkAction> {
+		const confirmations: (QuickPickItemOfT<StartWorkAction> | DirectiveQuickPickItem)[] = [
+			createQuickPickSeparator(fromNow(state.item.item.issue.updatedDate)),
+			createQuickPickItemOfT(
+				{
+					label: state.item.item.issue.title,
+					description: `${state.item.item.issue.repository?.owner ?? ''}/${
+						state.item.item.issue.repository?.repo ?? ''
+					}#${state.item.item.issue.id}`,
+					detail: state.item.item.issue.body ?? '',
+					iconPath:
+						state.item.item.issue.author?.avatarUrl != null
+							? Uri.parse(state.item.item.issue.author.avatarUrl)
+							: undefined,
+					buttons: [StartWorkQuickInputButton],
+				},
+				'start',
+			),
+			createDirectiveQuickPickItem(Directive.Noop, false, { label: '' }),
+			createQuickPickSeparator('Actions'),
+			createQuickPickItemOfT(
+				{
+					label: 'Start Work...',
+					detail: `Will start working on this issue`,
+				},
+				'start',
+			),
+		];
+
+		const step = this.createConfirmStep(
+			`Issue ${state.item.item.issue.repository?.owner ?? ''}/${state.item.item.issue.repository?.repo ?? ''}#${
+				state.item.item.issue.id
+			}`,
+			confirmations,
+			undefined,
+			{
+				placeholder: 'Choose an action to perform',
+				onDidClickItemButton: (_quickpick, button, _item) => {
+					switch (button) {
+						case StartWorkQuickInputButton:
+							if (isDirectiveQuickPickItem(_item)) return;
+							this.startWork(state);
+							return true;
+					}
+					return false;
+				},
+			},
+		);
+
+		const selection: StepSelection<typeof step> = yield step;
+		return canPickStepContinue(step, state, selection) ? selection[0].item : StepResultBreak;
+	}
+
+	private startWork(state: PartialStepState<State>, item?: StartWorkItem) {
+		state.action = 'start';
+		if (item != null) {
+			state.item = item;
+		}
+	}
+
+	private async getConnectedIntegrations(): Promise<Map<IntegrationId, boolean>> {
+		const connected = new Map<IntegrationId, boolean>();
+		await Promise.allSettled(
+			supportedStartWorkIntegrations.map(async integrationId => {
+				const integration = await this.container.integrations.get(integrationId);
+				connected.set(integrationId, integration.maybeConnected ?? (await integration.isConnected()));
+			}),
+		);
+
+		return connected;
+	}
+}
+
+async function updateContextItems(container: Container, context: Context) {
+	context.result = {
+		items:
+			(await container.integrations.getMyIssues([HostingIntegrationId.GitHub]))?.map(i => ({
```

Comment:
Perhaps we can do one of the following:
1. Create a `startWorkProvider`
2. Rename the `launchpadProvider` to be a bit more flexible outside the domain of Launchpad itself and extend it

And then we can add functions to the provider to:
1. Get searched issues, with caching and error handling like we do for Launchpad items, and the ability to filter out stale ones and other post-processing like we do with PRs for Launchpad (not that we have to implement any post-processing now, just the caching and error handling parts)
2. Complete actions on "start work" items, like creating a branch, etc.
3. Set up other utilities that we want in the future for Start Work so that the command file doesn't get out of hand.
4. Use those functions here as needed, like `launchpad` does with `launchpadProvider`.